S3Proxy/cache.go

```diff
@@ -5,40 +5,62 @@ import (
 	"time"
 )
 
-var s3Buckets = map[string]*s3BucketCacheItem{}
+var s3Buckets = map[string]*S3BucketCacheItem{}
+var s3Objects = map[string]*S3ObjectCacheItem{}
 
-type s3BucketCacheItem struct {
+// S3BucketCacheItem is a cache entry representing bucket info
+type S3BucketCacheItem struct {
```

Comment:
I have implemented the cache introducing the s3CacheItem struct (not exported). And I have embedded it into the Object and Bucket cache items. I'm just not convinced about the naming. Not sure if CacheItem makes the most sense.